pkg/k8s/apis/cilium.io/v2/types.go

```diff
@@ -158,3 +159,50 @@ type CiliumNetworkPolicyList struct {
 	// Items is a list of CiliumNetworkPolicy
 	Items []CiliumNetworkPolicy `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CiliumEndpoint is the status of a Cilium policy rule
+// +k8s:openapi-gen=false
+type CiliumEndpoint struct {
+	// +k8s:openapi-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	Status CiliumEndpointDetail `json:"status"`
```

Comment:
I'm sure you thought about the current and future of this. Our current model is both `Spec` and `Status`. I'm OK to make it read-only and then add `Spec` later on and allowing to modify certain elements such as Configuration or Labels`

Reply:
This was part of the intention (aanm mentioned that k8s merged a change that allows using a status that cannot be edited by a user). It was, originally, `endpoint` but that wasn't too helpful. `status` isn't ideal, though, because the endpoint struct we use (the one also returned in `cilium endpoint get`) has a `status` field inside it.
Anyway, it should extend in a reasonable way.

Reply:
I'm ok as-is

Reply:
Just FYI

>aanm mentioned that k8s merged a change that allows using a status that cannot be edited by a user

That was my understanding of that k8s merge regarding that feature. I was wrong. I've spent a couple hours trying to understand this behavior with k8s 1.11.

The `status` field in the CRD is writable by everyone. The `status` field on all other k8s objects is non-writable, in a sense k8s knows the `status` field should not be modified by external entities.

So, it's up to the controller of that CRD object, in our case cilium, to be aware of external changes of the `status` field and keep it updated as possible. We can still, and should, use `status` to report the `status` of a CRD.